### PR TITLE
QA:  PDF -> PNG+HTML

### PR DESCRIPTION
**Pull request by profxj**
Refactor code to create QA plots in PNG instead of PDF.
Plots are largely unmodified.

Also includes code to generate HTML files that 'ingest' the PNG files.

A few other odds and ends included.

### CODE REVIEW

**Comment by profxj** [comment]
I should add that a little more work will likely need to be
done for AMRED.

And things are passing the Development suite.

**Comment by rcooke-ast** [comment]
Is it worth adding (even minimal) docs about accessing the QA?

File: pypit/arqa.py [hunk 1]
```diff
@@ -514,6 +521,16 @@ def plot_orderfits(slf, model, ydata, xdata=None, xmodl=None, textplt="Slit", ma
...
       Value used in arrays to indicate a masked value
     """
+    # Outfil
+    module = inspect.stack()[0][3]
+    if 'Arc' in desc:
+        module += '_Arc'
+    elif 'Blaze' in desc:
+        module += '_Blaze'
+    else:
+        msgs.error("Should not get here")
```
Comment (rcooke-ast) [inline pypit/arqa.py]: Maybe use a slightly more informative error?

Or, perhaps this should raise a `msgs.bug` instead?

File: pypit/arqa.py [hunk 2]
```diff
@@ -715,10 +736,9 @@ def slit_profile(slf, mstrace, model, lordloc, rordloc, msordloc, textplt="Slit"
     return
 
 
-def slit_trace_qa(slf, frame, ltrace, rtrace, extslit, desc="", root='trace', outfil=None, normalize=True,
-                  use_slitid=None):
-    """
-    Generate a QA plot for the traces
+def slit_trace_qa(slf, frame, ltrace, rtrace, extslit, desc="",
+                  root='trace', outfil=None, normalize=True, use_slitid=None):
```
Comment (rcooke-ast) [inline pypit/arqa.py]: Seems like we don't need to pass in `outfil` anymore. If we still need it, maybe it's best to rename this to `outfile` for consistency?

File: pypit/armlsd.py [hunk 3]
```diff
@@ -81,6 +81,7 @@ def ARMLSD(fitsdict, reuseMaster=False, reloadMaster=True):
             arproc.get_datasec_trimmed(slf, fitsdict, det, scidx)
             # Setup
             setup = arsort.instr_setup(slf, det, fitsdict, setup_dict, must_exist=True)
+            slf.setup = setup
```
Comment (profxj) [inline pypit/armlsd.py]: added

File: pypit/armlsd.py [hunk 4]
```diff
@@ -210,8 +212,8 @@ def ARMLSD(fitsdict, reuseMaster=False, reloadMaster=True):
             # Using model sky, calculate a flexure correction
 
         # Close the QA for this object
-        if not msgs._debug['no_qa']:
-            slf._qa.close()
+        #if not msgs._debug['no_qa']:
```
Comment (profxj) [inline pypit/armlsd.py]: removed

and a few other related edits were made

File: pypit/armsgs.py [hunk 5]
```diff
@@ -153,17 +154,26 @@ def close(self):
         """
         Close the log file and QA PDFs before the code exits
         """
+        from pypit import arqa
         # Close PDFs
```
Comment (profxj) [inline pypit/armsgs.py]: I removed the closing of the PDF
But the import is needed to auto-make the HTML files.

File: pypit/arsave.py [hunk 6]
```diff
@@ -39,7 +39,7 @@ def save_arcids(fname, pixels):
             ans = ''
             while ans != 'y' and ans != 'n' and ans != 'r':
                 msgs.warn("File %s exists!" % (fname), verbose=settings.argflag['output']['verbosity'])
-                ans = raw_input(msgs.input()+"Overwrite? (y/n)")
+                ans = input(msgs.input()+"Overwrite? (y/n)")
```
Comment (profxj) [inline pypit/arsave.py]: Am leaving this one be as I don't think this
portion of code gets touched much.

File: pypit/arqa.py [hunk 7]
```diff
@@ -38,9 +47,12 @@ def arc_fit_qa(slf, fit, outfil=None, ids_only=False, title=None):
     outfil : str, optional
```
Comment (profxj) [inline pypit/arqa.py]: fixed

File: pypit/arqa.py [hunk 8]
```diff
@@ -431,32 +421,45 @@ def obj_trace_qa(slf, frame, ltrace, rtrace, objids, root='trace', outfil=None,
...
+    plt.close()
 
 
-def obj_profile_qa(slf, specobjs, scitrace):
+def obj_profile_qa(slf, specobjs, scitrace, det):
     """ Generate a QA plot for the object spatial profile
     Parameters
     ----------
     """
+    module = inspect.stack()[0][3]
```
Comment (profxj) [inline pypit/arqa.py]: Commented (in a few places).
It is just grabbing the name of the method.

Comment (profxj) [inline pypit/arqa.py]: Have replaced 'module' with 'method' throughout.

File: pypit/arqa.py (hunk 1 above)
Comment (profxj) [inline pypit/arqa.py]: Edited accordingly (and to a bug)

File: pypit/arqa.py (hunk 2 above)
Comment (profxj) [inline pypit/arqa.py]: Agreed.  Removed

**Comment by profxj** [comment]
Updated the qa.rst docs considerably.  This includes
ingesting a series of example QA files (in jpg format to 
limit their size).

**Comment by profxj** [comment]
Merging.

**Pull request merged**